### PR TITLE
Prepare public GVCFs for the combiner and VQSR. Add prep_inputs_for_combiner.py.

### PR DESCRIPTION
**Pull request by vladsaveliev**
* Add script `prep_inputs_for_combiner.py` to generate the `combiner_gvcfs.py` input sample maps
* Add `wdl/prepare-gvcfs.wdl` to post-process GVCFs from the public bucket `gs://fc-56ac46ea-efc4-4683-b6d5-6d95bed41c5e/CCDG_14151/Project_CCDG_14151_B01_GRM_WGS.gVCF.2020-02-12/Project_CCDG_14151_B01_GRM_WGS.gVCF.2020-02-12/` and make them VQSR-ready
* Rename Snakefile → prep_warp_inputs.smk
* Add input type `gvcf` for `prep_warp_inputs.smk`, and two rule: `make_prepare_gvcf_input` and `copy_gvcf` to prepare input for `wdl/prepare-gvcfs.wdl`.
* Some cleanup.

### CODE REVIEW

File: wdl/prepare-gvcfs.wdl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    output {
+        File output_vcf = outputPath
+        File output_vcf_index = outputPath + ".tbi"
+    }
+
+    runtime {
+        memory: "3 GB"
+        docker: "quay.io/biocontainers/bcftools:1.10.2--h4f4756c_2"
```
Comment (pdiakumis) [inline wdl/prepare-gvcfs.wdl:97]: I wonder if and what the costs are for pulling in images from Quay, DockerHub etc.

File: prep_inputs_for_combiner.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+
+"""
+Uploads GVCFs and QC files to CPG and senerates sample maps for CPG-QC
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:4]:
```suggestion
Uploads GVCFs and QC files to CPG and generates sample maps for CPG-QC
```
Also: is that CPG-QC meant to be joint-calling?

File: prep_inputs_for_combiner.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    'samples_ped',
+    help='Ped file with input sample names. If not provided, '
+         '`{datasets_dir}/{datasets_name}/samples.ped` is read.'
+)
+@click.option(
+    '--warp-executions-bucket',
+    'warp_executions_bucket',
+    default=('gs://playground-us-central1/cromwell/executions/'
+             'WGSMultipleSamplesFromBam'),
+    help='Bucket with WARP workflow outputs'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:111]: Sorry for nitpicking:

```suggestion
    help='Bucket with WARP workflow outputs.'
```

File: prep_inputs_for_combiner.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+)
+@click.option(
+    '--datasets-dir',
+    'datasets_dir',
+    default='datasets',
+    help='Output folder. Default is "datasets/"'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:117]:
```suggestion
    help='Output folder. Default is "datasets/".'
```

File: prep_inputs_for_combiner.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+)
+@click.option(
+    '--work-dir',
+    'work_dir',
+    help='Directory to store temporary files'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:122]:
```suggestion
    help='Directory to store temporary files.'
```

File: prep_inputs_for_combiner.py [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+)
+@click.option(
+    '--split-rounds',
+    'split_rounds',
+    is_flag=True,
+    help='Break samples into 2 groups to produce tests for the gVCF combiner'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:128]:
```suggestion
    help='Break samples into 2 groups to produce tests for the GVCF combiner.'
```

File: prep_inputs_for_combiner.py [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+)
+@click.option(
+    '--randomise-pop-labels',
+    'randomise_pop_labels',
+    is_flag=True,
+    help='Remove population labels for 1/3 of the samples to test sample-qc '
+         'ancestry detection'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:135]:
```suggestion
         'ancestry inference.'
```
Think inference is more accurate.

File: prep_inputs_for_combiner.py [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+)
+@click.option(
+    '--move-locally',
+    'move_locally',
+    is_flag=True,
+    help='Move GVCFs and picard files to the gs://cpg-fewgenomes-upload bucket'
```
Comment (pdiakumis) [inline prep_inputs_for_combiner.py:141]:
```suggestion
    help='Move GVCFs and picard files to the gs://cpg-fewgenomes-upload bucket.'
```

**Review by pdiakumis — APPROVED** [review]
Thank you Vlad, looks great! Mostly nitpicks and some typos.

File: datasets/50genomes-gvcf/50genomes-gvcf-gvcf.json [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG01084/HG01084.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG00512/HG00512.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/NA19127/NA19127.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/NA10839/NA10839.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/NA19173/NA19173.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG02074/HG02074.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG00429/HG00429.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG01261/HG01261.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/HG01074/HG01074.g.vcf.gz",
+        "gs://cpg-fewgenomes-temporary/original-gvcf/NA19983/g.g.vcf.gz"
```
[the remark this answers is not in the document]
Comment (pdiakumis) [reply datasets/50genomes-gvcf/50genomes-gvcf-gvcf.json:49]:
```
gsutil ls gs://cpg-fewgenomes-temporary/original-gvcf/NA19983/gvcf/
gs://cpg-fewgenomes-temporary/original-gvcf/NA19983/gvcf/NA19983.g.vcf.gz
gs://cpg-fewgenomes-temporary/original-gvcf/NA19983/gvcf/NA19983.g.vcf.gz.tbi
```

[the remark this answers is not in the document]
Comment (pdiakumis) [reply datasets/50genomes-gvcf/50genomes-gvcf-gvcf.json:49]: Actually, all those are in a `gvcf` subdirectory, unless I'm missing something.

[the remark this answers is not in the document]
Comment (vladsaveliev) [reply datasets/50genomes-gvcf/50genomes-gvcf-gvcf.json]: Fixed!

File: wdl/prepare-gvcfs.wdl [hunk 10]
```diff
@@ -0,0 +1,99 @@
+version 1.0
+
+# WORKFLOW DEFINITION
+
+workflow PrepareGvcfsWf {
+  input {
+    Array[File] input_gvcfs
+    File input_regions = "gs://cpg-reference/hg38/v0/noalt.bed"
```
[the remark this answers is not in the document]
Comment (vladsaveliev) [reply wdl/prepare-gvcfs.wdl:8]: Nope, prepared it myself!

File: wdl/prepare-gvcfs.wdl (hunk 1 above)
Comment (vladsaveliev) [reply wdl/prepare-gvcfs.wdl:97]: Images should be cached so shouldn't be expensive. Also I'm moving this stuff into a Hail Batch pipeline, which is definitely good in caching images. Longer-term, will move these images to Australia.

**Comment by vladsaveliev** [comment]
Somehow forgot to respond to your review, Peter! Thanks so much for the comments and edits. Merged now.

**Pull request merged**